ts/main.ts

```diff
@@ -0,0 +1,64 @@
+interface Anime {
+  mal_id: number;
+  title: string;
+  episodes: number;
+  images: {
+    jpg: {
+      image_url: string;
+    };
+  };
+  [key: string]: any; // To account for any additional fields
+}
+
+const animeContainer = document.querySelector(
+  '.anime-container',
+) as HTMLDivElement;
+
+async function fetchAnime(): Promise<Anime[]> {
+  try {
+    const response = await fetch('https://api.jikan.moe/v4/top/anime');
+    const data = await response.json();
+    return data.data as Anime[];
```

Comment:
After you make the `fetch` request you want to check if the `response` is `ok` or else you may be trying to continue with a `rejected` promise information.
```
const response = await fetch('https://api.jikan.moe/v4/top/anime');
if (!response.ok) throw new Error('Fetch for top anime failed!');
const data = await response.json()
...
```